src/content_mapblock.cpp

```diff
@@ -77,17 +77,20 @@ void MapblockMeshGenerator::useTile(int index, bool disable_backface_culling)
 {
 	tile = getNodeTileN(n, p, index, data);
 	if (!data->m_smooth_lighting)
-		color = encode_light_and_color(light, tile.color, f->light_source);
+		color = encode_light(light, f->light_source);
 	if (disable_backface_culling)
-		tile.material_flags &= ~MATERIAL_FLAG_BACKFACE_CULLING;
-	tile.material_flags |= MATERIAL_FLAG_CRACK_OVERLAY;
+		for (u8 layer = 0; layer < MAX_TILE_LAYERS; layer++)
+			tile.layers[layer].material_flags &=
+				~MATERIAL_FLAG_BACKFACE_CULLING;
+	for (u8 layer = 0; layer < MAX_TILE_LAYERS; layer++)
+		tile.layers[layer].material_flags |= MATERIAL_FLAG_CRACK_OVERLAY;
```

Comment:
I don’t demand but would be glad to see if you
1. place conditionals under loops, as the code becomes messy very quickly otherwise,
2. use `int` for loop counters: that looks cleaner (IMO), and the actual size doesn’t matter at all.